src/main/java/com/google/maps/GeoApiContext.java

```diff
@@ -141,9 +141,9 @@ public GeoApiContext(RequestHandler requestHandler) {
         config.supportsClientId, query.toString());
   }
 
-  <T, R extends ApiResponse<T>> PendingResult<T> post(ApiConfig config,
+  <T, R extends ApiResponse<T>, P> PendingResult<T> post(ApiConfig config,
```

Comment:
Apologies for not being misleading here - by 'generic' I meant applicable across different APIs, not necessarily generic in the type sense.

This type parameter doesn't add any type safety (which would be the purpose of adding it). I think it would be less confusing to use a bare `Object`, as that's what this unbounded type param is doing anyway. Alternatively this method could accept the serialized payload as a String or byte array (which would theoretically allow other APIs to POST non-JSON data, though there are none of them supported in this library so I wouldn't stress about that). `PendingResultBase` could then store the bare Object or bytes and your API request class could do the appropriate serialization and pass it through to `PendingResultBase` in a new method analogous to `param(String, String)`.

I think this strategy makes sense as the method for encoding POST data can vary from API to API, where as the method for encoding GET data is well-defined in the HTTP spec, so applies uniformly across APIs.

Does this make sense?